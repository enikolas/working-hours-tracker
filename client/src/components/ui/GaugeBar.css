.gauge-bar {
  background: #fff;
  margin-bottom: 2px;
  border: 1px solid #5a717f;
}

.gauge-bar .wrapper {
  height: 10px;
  background: #c5ccd1;
  width: 155px;
}

.gauge-bar .marks {
  border: 2px solid #254356;
  border-top: none;
  border-bottom: none;
  height: 10px;
  margin-right: 50px;
  position: absolute;
  width: 155px;
}

.gauge-bar .bar {
  height: 10px;
  background: #909fa8;
  border-radius: 2px;
  transition: ease-in width 0.3s;
}

.gauge-bar .bar.debt {
  background: #d0808b;
}

.gauge-bar .bar.surplus {
  background: #86ae80;
}
