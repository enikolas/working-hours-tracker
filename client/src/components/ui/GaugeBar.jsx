import React from 'react';
import PropTypes from 'prop-types';

import './GaugeBar.css';

const GaugeBar = ({ currentValue, referenceValue, verticalBar }) => {
  const percentage = (currentValue / referenceValue) * 100;
  const surplusClass = percentage > 100 ? ' surplus' : '';
  const debtClass = percentage < 100 ? ' debt' : '';
  const stylePropToChange = verticalBar ? 'height' : 'width';
  return (
    <div className="gauge-bar">
      <div className="wrapper">
        <div className="marks" />
        <div
          className={`bar${surplusClass}${debtClass}`}
          style={{ [stylePropToChange]: `${percentage}%` }}
        />
      </div>
    </div>
  );
};

GaugeBar.propTypes = {
  currentValue: PropTypes.number.isRequired,
  referenceValue: PropTypes.number.isRequired,
  verticalBar: PropTypes.bool
};

GaugeBar.defaultProps = {
  verticalBar: false
};

export default GaugeBar;
