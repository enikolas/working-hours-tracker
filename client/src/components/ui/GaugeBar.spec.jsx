import React from 'react';
import { shallow } from 'enzyme';

import GaugeBar from './GaugeBar';

describe('GaugeBar', () => {
	it('shoul render a surplus gauge if current > ref', () => {
		const wrapper = shallow(<GaugeBar
			currentValue={100}
			referenceValue={10}
		/>);
		expect(wrapper).toMatchSnapshot();
	});
	it('shoul render a debt gauge if current > ref and vertical', () => {
		const wrapper = shallow(<GaugeBar
			currentValue={10}
			referenceValue={100}
			verticalBar
		/>);
		expect(wrapper).toMatchSnapshot();
	});
});
