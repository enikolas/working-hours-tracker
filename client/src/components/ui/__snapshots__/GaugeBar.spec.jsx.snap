// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GaugeBar shoul render a debt gauge if current > ref and vertical 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GaugeBar
    currentValue={10}
    referenceValue={100}
    verticalBar={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="wrapper"
      >
        <div
          className="marks"
        />
        <div
          className="bar debt"
          style={
            Object {
              "height": "10%",
            }
          }
        />
      </div>,
      "className": "gauge-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="marks"
          />,
          <div
            className="bar debt"
            style={
              Object {
                "height": "10%",
              }
            }
          />,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "marks",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bar debt",
            "style": Object {
              "height": "10%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="wrapper"
        >
          <div
            className="marks"
          />
          <div
            className="bar debt"
            style={
              Object {
                "height": "10%",
              }
            }
          />
        </div>,
        "className": "gauge-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="marks"
            />,
            <div
              className="bar debt"
              style={
                Object {
                  "height": "10%",
                }
              }
            />,
          ],
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "marks",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bar debt",
              "style": Object {
                "height": "10%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GaugeBar shoul render a surplus gauge if current > ref 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GaugeBar
    currentValue={100}
    referenceValue={10}
    verticalBar={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="wrapper"
      >
        <div
          className="marks"
        />
        <div
          className="bar surplus"
          style={
            Object {
              "width": "1000%",
            }
          }
        />
      </div>,
      "className": "gauge-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="marks"
          />,
          <div
            className="bar surplus"
            style={
              Object {
                "width": "1000%",
              }
            }
          />,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "marks",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bar surplus",
            "style": Object {
              "width": "1000%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="wrapper"
        >
          <div
            className="marks"
          />
          <div
            className="bar surplus"
            style={
              Object {
                "width": "1000%",
              }
            }
          />
        </div>,
        "className": "gauge-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="marks"
            />,
            <div
              className="bar surplus"
              style={
                Object {
                  "width": "1000%",
                }
              }
            />,
          ],
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "marks",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bar surplus",
              "style": Object {
                "width": "1000%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
